examples/tv-app/linux/AppImpl.cpp

```diff
@@ -0,0 +1,219 @@
+    DECLARE_DYNAMIC_CLUSTER(ZCL_APPLICATION_BASIC_CLUSTER_ID, accountLoginAttrs),
+    DECLARE_DYNAMIC_CLUSTER(ZCL_APPLICATION_BASIC_CLUSTER_ID, contentLauncherAttrs),
+    DECLARE_DYNAMIC_CLUSTER(ZCL_APPLICATION_BASIC_CLUSTER_ID, mediaPlaybackAttrs),
+    DECLARE_DYNAMIC_CLUSTER(ZCL_APPLICATION_BASIC_CLUSTER_ID, targetNavigatorAttrs),
+    DECLARE_DYNAMIC_CLUSTER(ZCL_APPLICATION_BASIC_CLUSTER_ID, channelAttrs) DECLARE_DYNAMIC_CLUSTER_LIST_END;
+
+// Declare Content App endpoint
+DECLARE_DYNAMIC_ENDPOINT(contentAppEndpoint, contentAppClusters);
+
+ContentAppImpl::ContentAppImpl(const char * szVendorName, uint16_t vendorId, const char * szApplicationName, uint16_t productId,
+                               const char * szApplicationVersion)
+{
+    mApplicationBasic.SetApplicationName(szApplicationName);
+    mApplicationBasic.SetVendorName(szApplicationName);
+    mApplicationBasic.SetVendorId(vendorId);
+    mApplicationBasic.SetProductId(productId);
+    mApplicationBasic.SetApplicationVersion(szApplicationVersion);
+}
+
+void ApplicationBasicImpl::SetApplicationName(const char * szApplicationName)
+{
+    ChipLogProgress(DeviceLayer, "ApplicationBasic[%s]: Application Name=\"%s\"", szApplicationName, szApplicationName);
+
+    strncpy(mApplicationName, szApplicationName, sizeof(mApplicationName));
+}
+
+void ApplicationBasicImpl::SetVendorName(const char * szVendorName)
+{
+    ChipLogProgress(DeviceLayer, "ApplicationBasic[%s]: Vendor Name=\"%s\"", mApplicationName, szVendorName);
+
+    strncpy(mVendorName, szVendorName, sizeof(mVendorName));
+}
+
+void ApplicationBasicImpl::SetApplicationVersion(const char * szApplicationVersion)
+{
+    ChipLogProgress(DeviceLayer, "ApplicationBasic[%s]: Application Version=\"%s\"", mApplicationName, szApplicationVersion);
+
+    strncpy(mApplicationVersion, szApplicationVersion, sizeof(mApplicationVersion));
+}
+
+bool AccountLoginImpl::Login(const char * tempAccountId, uint32_t setupPin)
+{
+    ChipLogProgress(DeviceLayer, "AccountLogin: Login TempAccountId=\"%s\" SetupPIN=\"%d\"", tempAccountId, setupPin);
+    return (setupPin == mSetupPIN);
+}
+
+uint32_t AccountLoginImpl::GetSetupPIN(const char * tempAccountId)
+{
+    ChipLogProgress(DeviceLayer, "AccountLogin: GetSetupPIN TempAccountId=\"%s\" returning setuppin=%d", tempAccountId, mSetupPIN);
+    return mSetupPIN;
+}
+
+ContentAppFactoryImpl::ContentAppFactoryImpl()
+{
+    mContentApps[1].GetAccountLogin()->SetSetupPIN(34567890);
+    mContentApps[2].GetAccountLogin()->SetSetupPIN(20202021);
+}
+
+ContentApp * ContentAppFactoryImpl::LoadContentAppByVendorId(uint16_t vendorId)
+{
+    for (unsigned int i = 0; i < sizeof(mContentApps); i++)
+    {
+        ContentAppImpl app = mContentApps[i];
+        if (app.GetApplicationBasic()->GetVendorId() == vendorId)
+        {
+            AppPlatform::GetInstance().AddContentApp(&app, &contentAppEndpoint, DEVICE_TYPE_CONTENT_APP);
```

Comment:
This is passing a pointer to a stack-allocated data structure.  As soon as this function returns, that becomes a dangling pointer.  It looks like the callee stores the pointer, not a copy of the data pointed to.

What am I missing here that makes this not a critical security bug?

Reply:
this was fixed in https://github.com/project-chip/connectedhomeip/pull/12634